cognite/client/_api/data_sets.py

```diff
@@ -0,0 +1,247 @@
+from typing import *
+
+from cognite.client import utils
+from cognite.client._api_client import APIClient
+from cognite.client.data_classes import DataSet, DataSetFilter, DataSetList, DataSetUpdate, TimestampRange
+
+
+class DataSetsAPI(APIClient):
+    _RESOURCE_PATH = "/datasets"
+    _LIST_CLASS = DataSetList
+
+    def __call__(
+        self,
+        chunk_size: int = None,
+        metadata: Dict[str, str] = None,
+        created_time: Union[Dict[str, Any], TimestampRange] = None,
+        last_updated_time: Union[Dict[str, Any], TimestampRange] = None,
+        external_id_prefix: str = None,
+        write_protected: bool = None,
+        limit: int = None,
+    ) -> Generator[Union[DataSet, DataSetList], None, None]:
+        """Iterate over data sets
+
+        Fetches data sets as they are iterated over, so you keep a limited number of data sets in memory.
+
+        Args:
+            chunk_size (int, optional): Number of data sets to return in each chunk. Defaults to yielding one data set a time.
+            metadata (Dict[str, str]): Custom, application-specific metadata. String key -> String value.
+            created_time (Union[Dict[str, Any], TimestampRange]): Range between two timestamps.
+            last_updated_time (Union[Dict[str, Any], TimestampRange]): Range between two timestamps.
+            external_id_prefix (str): Filter by this (case-sensitive) prefix for the external ID.
+            write_protected (bool): No description.
```

Comment:
We should have a description here.

Reply:
it came from swagger spec, so we have to go and comment on service contracts when we have a pr there

Reply:
>Specify whether the filtered data sets are write-protected, or not. Set to True to only list write-protected data sets.